### PR TITLE
[BUILD] Add support of external mvn with default mvn jvm opts when making distribution

### PR DESCRIPTION
**Pull request by goldenbean**
### _Why are the changes needed?_

most of time, we use the mvn pre-installed in the OS 

### _How was this patch tested?_

After running the script `build/dist`, it should echo the 'MVN' location before building the release package

```bash
build/dist --tgz --name release-demo \
        --spark-provided --flink-provided \
        --mvn $(which mvn) \
        -DskipTests \
        -Pjava-8 -Pspark-hadoop-3.2 -Pspark-3.0
```

### CODE REVIEW

File: build/dist [hunk 1]
```diff
@@ -68,6 +68,10 @@ while (( "$#" )); do
     --spark-provided)
       SPARK_PROVIDED=true
       ;;
+    --mvn)
```
Comment (pan3793) [inline build/dist:71]: let's also update usage

**Comment by pan3793** [comment]
> most of time, we use the mvn pre-installed in the OS

FYI, the `build/mvn` will fallback to `mvn` command if your pre-installed mvn version equals or greater than `maven.version` defined in `pom.xml` rather than download new one. We don't recommand use the lower maven version.

**Review by yaooqinn — APPROVED** [review]

File: build/dist (hunk 1 above)
Comment (goldenbean) [reply build/dist:71]: sure

File: build/dist [hunk 2]
```diff
@@ -41,12 +41,13 @@ function usage {
   echo ""
   echo "Usage:"
   echo "+--------------------------------------------------------------------------------------+"
-  echo "| ./build/dist [--name <custom_name>] [--tgz] [--flink-provided] [--spark-provided] <maven build options> |"
+  echo "| ./build/dist [--name <custom_name>] [--tgz] [--flink-provided] [--spark-provided] [--mvn <maven_executable>] <maven build options> |"
```
Comment (pan3793) [inline build/dist:44]: Let's break it into multi-lines, and keep it a good format

**Comment by goldenbean** [comment]
> > most of time, we use the mvn pre-installed in the OS
> 
> FYI, the `build/mvn` will fallback to `mvn` command if your pre-installed mvn version equals or greater than `maven.version` defined in `pom.xml` rather than download new one. We don't recommand to use the lower maven version.

Hi @pan3793 ,  thank you for the comments. We currently use only the maven 3.6.3 to build Kyuubi, and deploy in one of the staging environments. It seems fine for now. :) 

If there were some issues, I will follow the recommendation.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks @goldenbean, merging to master. And for maven versions, we encounter some Scala compile issues and maven shade plugin issue in past, and sometimes bump maven version can solve.

**Pull request closed without merging**